src/R/Editor/Impl/Signatures/SignatureHelp.cs

```diff
@@ -163,10 +163,13 @@ public partial class SignatureHelp : ISignature {
                 IREditorDocument document = REditorDocument.TryFromTextBuffer(SubjectBuffer);
                 if (document != null) {
                     SnapshotPoint? p = REditorDocument.MapCaretPositionFromView(TextView);
+                    var textView = TextView; // Cache so it doesn't go away when callback is invoked.
```

Comment:
Yes. We need to re-trigger session when information fetching is done. Basically, core editor is synchronous. It calls for parameters help and we need time. So we tell it that we don't have anything and then, when information comes, trigger signature help again. 

Hugues has slower laptop this is why it doesn't repro always. The calling object is still available on faster machines.